app/controllers/cinema_halls_controller.rb

```diff
@@ -14,16 +14,22 @@ def show
   def create
     cinema_hall = CinemaHall.new(cinema_hall_params)
     if [20, 50, 100, 200].include? cinema_hall.size
-      render json: cinema_hall, status: :created if cinema_hall.save
+      if cinema_hall.save
+        render json: cinema_hall, status: :created
+        CinemaHallsService.new(cinema_hall).assign_seat_names
+      end
```

Comment:
The best solution is to have whole logic inside the service. Consider that maybe in the future you will be able to have all this code inside this service and just return results or errors?

It's not topic for today, just consider it in future :)

Reply:
Will do, thank you